### PR TITLE
String literal with primitive concatenation

### PR DESCRIPTION
**Pull request by lunalunaa**
As titled.

### CODE REVIEW

File: base/src/main/java/org/aya/core/def/PrimDef.java [hunk 1]
```diff
@@ -208,7 +246,10 @@ public enum ID {
     /** Short for <em>Arend coe</em>. */
     ARCOE("arcoe"),
     SQUEEZE_LEFT("squeezeL"),
-    INVOL("invol");
+    INVOL("invol"),
+    STR("Str"),
```
Comment (imkiva) [inline base/src/main/java/org/aya/core/def/PrimDef.java:250]: Suggestion: change the type name to `String`. `Str` is just so ugly

**Review by imkiva — COMMENTED** [review]
LGTM

File: base/src/main/java/org/aya/core/def/PrimDef.java (hunk 1 above)
Comment (ice1000) [reply base/src/main/java/org/aya/core/def/PrimDef.java:250]: I agree, you can always use a renaming open to use a shorter name locally

File: base/src/main/java/org/aya/tyck/LittleTyper.java [hunk 2]
```diff
@@ -77,6 +78,7 @@ public record LittleTyper(@NotNull TyckState state, @NotNull LocalCtx localCtx)
       case FormTerm.Univ univ -> new FormTerm.Univ(univ.lift() + 1);
       case FormTerm.Interval interval -> new FormTerm.Univ(0);
       case PrimTerm.End end -> FormTerm.Interval.INSTANCE;
+      case PrimTerm.Str str -> state.primFactory().getOption(PrimDef.ID.STR).get().result();
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/LittleTyper.java:81]: Are you sure it's `.get().result()` not `getCall(ID.STR)`?

File: cli/src/main/java/org/aya/cli/parse/AyaProducer.java [hunk 3]
```diff
@@ -270,7 +270,8 @@ public QualifiedID visitQualifiedId(AyaParser.QualifiedIdContext ctx) {
       throw new ParsingInterruptedException();
     }
     var string = ctx.STRING();
-    if (string != null) return new Expr.LitStringExpr(pos, string.getText());
+    if (string != null)
+      return new Expr.LitStringExpr(pos, string.getText().substring(1, string.getText().length() - 1));
```
Comment (ice1000) [inline cli/src/main/java/org/aya/cli/parse/AyaProducer.java:274]: Escape

File: base/src/main/java/org/aya/tyck/LittleTyper.java (hunk 2 above)
Comment (lunalunaa) [reply base/src/main/java/org/aya/tyck/LittleTyper.java:81]: No it's just I forgot.

File: cli/src/main/java/org/aya/cli/parse/AyaProducer.java (hunk 3 above)
Comment (lunalunaa) [reply cli/src/main/java/org/aya/cli/parse/AyaProducer.java:274]: I'm not sure what this means. Should I do `translateEscapes()` here?

Comment (ice1000) [reply cli/src/main/java/org/aya/cli/parse/AyaProducer.java:274]: image

**Pull request merged**